### PR TITLE
[AKS] aks create/update: add --vnet-subnet_id validationChenyzh vnet validator

### PR DESCRIPTION
**Pull request by Czhang0727**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### CODE REVIEW

**Comment by Czhang0727** [comment]
> If this PR is for a new extension or change to an existing extension, use the following to try out the changes in this PR:
> 
> ```
> docker run -it microsoft/azure-cli:latest
> export EXT=<NAME>
> pip install --upgrade --target ~/.azure/cliextensions/$EXT "git+https://github.com/Czhang0727/azure-cli-extensions.git@chenyzh_vnetValidator#subdirectory=src/$EXT&egg=$EXT"
> ```

Tested in docker, expected result received.

bash-5.0# az aks create -g chenyzh --vnet-subnet-id dumy -n cztest --ssh-key-value .ssh/id_rsa.pub 
The behavior of this command has been altered by the following extension: aks-preview
--vnet-subnet-id is not a valid Azure resource ID.

bash-5.0# az aks create -g chenyzh --vnet-subnet-id /subscriptions/YourSubscriptionId/resourceGroups/YourResourceGroupName/providers/Microsoft.Network/virtualNetworks/Name -n cztest --ssh-key-value ~/.ssh/id_rsa.pub
The behavior of this command has been altered by the following extension: aks-preview
Resource group 'chenyzh' could not be found.

bash-5.0# az aks create -g chenyzh --vnet-subnet-id "" -n cztest --ssh-key-value ~/.ssh/id_rsa.pub
The behavior of this command has been altered by the following extension: aks-preview
Resource group 'chenyzh' could not be found.

**Comment by Czhang0727** [comment]
Did not realize "close and comment" will actually close entire PR while Azure DevOps just close the comment, re-open for continuing the review.

**Comment by Czhang0727** [comment]
Re-try validation as one of check is expire

**Comment by yonzhan** [comment]
add to S165.

**Comment by yonzhan** [comment]
add to S165.

**Comment by MyronFanQiu** [comment]
@Czhang0727 Hello. I remembered that I merged a same PR. Is that in CLI main repo?

**Comment by Czhang0727** [comment]
> @Czhang0727 Hello. I remembered that I merged the same PR. Is that in CLI main repo?

@MyronFanQiu  Hi, I already checked in the same update in azure-cli as recommended. The update is already checked in.
https://github.com/Azure/azure-cli/pull/12038

I added a little bouns as I saw there is a styling warning with an un-necessary return.

**Comment by Czhang0727** [comment]
@qwordy, any chance we can merge the change?

File: src/aks-preview/azext_aks_preview/_params.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def load_arguments(self, _):
         c.argument('no_ssh_key', options_list=['--no-ssh-key', '-x'])
         c.argument('pod_cidr')
         c.argument('service_cidr')
-        c.argument('vnet_subnet_id')
+        c.argument('vnet_subnet_id', type=str, validator=validate_vnet_subnet_id)
```
Comment (qwordy) [inline src/aks-preview/azext_aks_preview/_params.py]: Does it has a help message?

Comment (Czhang0727) [reply src/aks-preview/azext_aks_preview/_params.py]: yes it does, it keeps same as it was to be in _help.py

```
        - name: --vnet-subnet-id
          type: string
          short-summary: The ID of a subnet in an existing VNet into which to deploy the cluster.
```

**Comment by Czhang0727** [comment]
merge conflict for check-in

**Comment by Czhang0727** [comment]
@qwordy Hi, would it possible to have the file merged and checked in?
I found this PR is stuck here for too long, and new PRs are keep conflicts the fixing there.

**Comment by qwordy** [comment]
We are very sorry to leave this PR unmerged for a long time. Can you resolve the conflicts again?

**Comment by Czhang0727** [comment]
@qwordy  Resolved conflict and verified through the pipeline.
Let me know if there is still any blocker for check in

**Pull request merged**